Trail Counters to Production

Charlie-Henry opened the pull request:
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/12453

## Associated repo
https://github.com/cityofaustin/atd-trail-counter-data

## Testing

**Steps to test:**
1. git pull this branch down to your local machine
2. `docker compose build` this version of the airflow stack
3.  `docker compose run --rm airflow-cli dags test atd_trail_counters` will run the DAG
4. View the logs and verify it's pulling data from the ecocounter website something like:
```
DEBUG:counter_data.py:0 records found for Shoal Creek Blvd at 38th St PC
DEBUG:counter_data.py:0 records found for Southern Walnut Creek Trail - Urban Trail
...
```


---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Add note to 1PW secrets moved to API vault and check for duplicates

Charlie-Henry committed:
first dag

Charlie-Henry committed:
requested changes

Charlie-Henry committed:
Merge branch 'airflow-v2' into ch-trail-counters

Charlie-Henry committed:
Merge pull request #103 from cityofaustin/ch-trail-counters

Trail Counters Data Publishing DAG

Charlie-Henry committed:
Trail Counters prev execution date

Charlie-Henry commented on dags/ped_data_pub.py:
```diff
@@ -0,0 +1,80 @@
+    catchup=False,
+) as dag:
+
+    @task(
+        task_id="get_env_vars",
+        execution_timeout=timedelta(seconds=30),
+    )
+    def get_env_vars():
+        from utils.onepassword import load_dict
+
+        env_vars = load_dict(REQUIRED_SECRETS)
+        return env_vars
```
Copying John's pattern from here:
https://github.com/cityofaustin/atd-airflow/pull/116

johnclary commented on dags/ped_data_pub.py:
```diff
@@ -0,0 +1,80 @@
+
+    env_vars = get_env_vars()
+
+    start = "{{ prev_start_date_success.strftime('%Y-%m-%d') if prev_start_date_success or '1970-01-01'}}"
```
this was erroring for me until i changed `or` to `else`:

```
[2023-07-12, 13:59:53 UTC] {abstractoperator.py:597} ERROR - Exception rendering Jinja template for task 'trail_counter_data_publish', field 'command'. Template: "python counter_data.py --start {{ prev_start_date_success.strftime('%Y-%m-%d') if prev_start_date_success or '1970-01-01'}}"
Traceback (most recent call last):
....
jinja2.exceptions.UndefinedError: 'None' has no attribute 'strftime'
```

johnclary commented on dags/ped_data_pub.py:
```diff
@@ -0,0 +1,80 @@
+}
+
+with DAG(
+    dag_id="atd_trail_counters",
```
it'd be nice to have this DAG's file name be more similar to the DAG name. e.g. `atd_trail_counters.py`

johnclary reviewed (dismissed):
@Charlie-Henry this is looking great. just a few minor changes, and I'm going to start a Slack thread wrt to Socrata entries in 1password.

When I tested the code, it did process a bunch of data. Is that ok? On that note, would you mind adding a bit more detail to this repo's [readme](https://github.com/cityofaustin/atd-trail-counter-data)? It wasn't quickly obvious from the docs what this script was going to do, if we need to worry about duplicating data, and where the public dataset can be found.

frankhereford commented on dags/ped_data_pub.py:
```diff
@@ -0,0 +1,80 @@
+# test locally with: docker compose run --rm airflow-cli dags test atd_trail_counters
+
+import os
```
It looks like this isn't used, so I think it can go.

frankhereford commented on dags/ped_data_pub.py:
```diff
@@ -0,0 +1,80 @@
+from utils.slack_operator import task_fail_slack_alert
+from utils.onepassword import load_dict
+
+from onepasswordconnectsdk.client import Client, new_client
```
Same with these, I think these are unused.

frankhereford commented on dags/ped_data_pub.py:
```diff
@@ -0,0 +1,80 @@
+from airflow.operators.docker_operator import DockerOperator
+
+from utils.slack_operator import task_fail_slack_alert
+from utils.onepassword import load_dict
```
It looks like you're importing this in a function's namespace below, so I think this can be removed.

frankhereford reviewed (approved):
I agree with John's suggestions, esp with the bugfix and the filename. I was able to get this running, and I don't have anything valuable to add beyond a couple of imports that can get removed. This is great Charlie, and thanks for cleaning up this branch so we didn't loose your work!!

🚢 🚢 🚢

Charlie-Henry commented:
@johnclary 
> it did process a bunch of data. Is that ok?

Yes, likely used yesterday's date in the `--start` param and pulled in that data. Each row is protected by an ID so no duplication issues should arise. Run this script to your heart's content!

Charlie-Henry committed:
rename and requested changes

Charlie-Henry commented:
Since the file was renamed so lots of deltas here's what I addressed with my commit:
- Dropped unnecessary imports thx @frankhereford 
- `else` instead of `or` in the start param
-  Made a trail counters 1pass entry where the only item is the Socrata resource ID and is now pointing to that instead

[Dataset is here](https://datahub.austintexas.gov/Transportation-and-Mobility/Trail-Counters-Daily-Totals/26tt-cp67) internal because [blocked](https://github.com/cityofaustin/atd-data-tech/issues/8642)

Charlie-Henry committed:
adding force_pull to docker

Charlie-Henry committed:
undoing changes to onepass util and gitignore

Charlie-Henry committed:
incorporating feedback from the other PR
